### PR TITLE
feat: migrate to CDK v2

### PR DESCRIPTION
**Pull request by sobolk**
This PR migrates `amplify-cli-export-construct` to CDK v2.

Highlights:
- CDK v1 replaced with 2.47.0
- Only appsync L1 constructs were needed, they're part of `aws-cdk-lib` . No need to use alpha L2 constructs
- I had to bump `projen` and `jsii` due to compatibility issues, see https://github.com/aws/aws-cdk/issues/21514, https://github.com/projen/projen/issues/2024
- Fixed integration tests. Baselined on fixed CLI 10.0.0 version instead of random beta tag.

Testing:
- unit tests pass
- integration tests pass

image

### CODE REVIEW

File: .github/workflows/integration-test.yml [hunk 1]
```diff
@@ -21,8 +21,8 @@ jobs:
           node-version: 14.17.6
```
Comment (akshbhu) [inline .github/workflows/integration-test.yml:21]: not part of this , but we can upgrade node to 16 LTS

**Review by akshbhu — COMMENTED** [review]
Nice one on tests !!!
some minor nits , otherwise lgtm!!!

File: package.json [hunk 2]
```diff
@@ -35,8 +35,8 @@
     "organization": true
   },
   "devDependencies": {
-    "@aws-cdk/assert": "^1.162.0",
-    "@aws-cdk/core": "^1.162.0",
+    "aws-cdk-lib": "2.47.0",
```
[the remark this answers is not in the document]
Comment (sobolk) [reply package.json:38]: I tried that initially but it gave me this warning
```
warning JSII6: A "peerDependency" on "aws-cdk-lib" at "^2.47.0" means you should take a "devDependency" on "aws-cdk-lib" at "2.47.0" (found "^2.47.0")
```

File: package.json [hunk 3]
```diff
@@ -45,41 +45,24 @@
...
-    "@aws-cdk/aws-cloudformation": "^1.162.0",
-    "@aws-cdk/aws-cognito": "^1.162.0",
-    "@aws-cdk/aws-iam": "^1.162.0",
-    "@aws-cdk/aws-lambda": "^1.162.0",
-    "@aws-cdk/aws-s3": "^1.162.0",
-    "@aws-cdk/aws-s3-assets": "^1.162.0",
-    "@aws-cdk/aws-s3-deployment": "^1.162.0",
-    "@aws-cdk/cloudformation-include": "^1.162.0",
-    "constructs": "^3.4.39"
+    "aws-cdk-lib": "^2.47.0",
```
[the remark this answers is not in the document]
Comment (sobolk) [reply package.json:60]: Also. This is the layout `projen` initially generated.

File: .github/workflows/integration-test.yml (hunk 1 above)
Comment (sobolk) [reply .github/workflows/integration-test.yml:21]: The package's baseline is 14. I.e. it means it runs on 14+. So we should keep this as is until we move baseline up.

File: src/base-exported-backend.ts [hunk 4]
```diff
@@ -75,28 +73,28 @@ export class BaseAmplifyExportedBackend extends Construct {
...
           }
           return obj;
-        }, {}) : 
+        }, {}) :
         cfnIncludeProps.loadNestedStacks
     };
-    
-    
-    
+
```
[the remark this answers is not in the document]
Comment (sobolk) [reply src/base-exported-backend.ts:95]: Removed. (visible in new diff, but somehow didn't make this comment outdated)

File: package.json (hunk 3 above)
[the remark this answers is not in the document]
Comment (akshbhu) [reply package.json:60]: Oh these are project generated.

Can we use ~ or ^ in projen config files, just curious ?

**Review by akshbhu — APPROVED** [review]

File: package.json (hunk 3 above)
[the remark this answers is not in the document]
Comment (sobolk) [reply package.json:60]: Normally this is the only file that gets edited https://github.com/aws-amplify/amplify-cli-export-construct/blob/fa1c2bcbfabd681c6a832419c739073d0f4226a9/.projenrc.js#L10 . It does not allow ^ there, but it seems to use ^ in generated files.

In practice I had to re-generate and cherry pick changes related to versions from generator output and not touch GH workflow files as they were manually adjusted after initial generation (except integ tests).

**Comment by dreamorosi** [comment]
This is great, is there an issue we can follow to track progress about when it'll ship?

**Comment by sobolk** [comment]
> This is great, is there an issue we can follow to track progress about when it'll ship?
I just shipped 0.0.6 version.

**Pull request merged**